source/Camera/Maintenance/CorrectCameraTrueness/CorrectCameraTrueness.cpp

```diff
@@ -0,0 +1,119 @@
+correction, and optionally saving that new correction to the camera.
+
+The correction will persist on the camera even though the camera is power-cycled or
+connected to a different PC. After saving a correction, it will automatically be used any
+time that camera captures a new point cloud.
+*/
+
+#include <Zivid/Zivid.h>
+
+#include <algorithm>
+#include <iomanip>
+#include <iostream>
+#include <vector>
+
+namespace
+{
+    bool yesNoPrompt(const std::string &question)
+    {
+        while(true)
+        {
+            std::string response;
+            std::cout << question << ": y/n " << std::endl;
+            std::cin >> response;
+
+            if(std::cin.fail() || response == "n" || response == "N")
+            {
+                return false;
+            }
+            if(response == "y" || response == "Y")
+            {
+                return true;
+            }
+            std::cout << "Invalid response. Please respond with either 'y' or 'n'." << std::endl;
+        }
+    }
+
+    std::vector<Zivid::Experimental::Calibration::InfieldCorrectionInput> collectDataset(Zivid::Camera &camera)
+    {
+        std::vector<Zivid::Experimental::Calibration::InfieldCorrectionInput> dataset;
+        std::cout << "Please point the camera at a Zivid in-field calibration object. " << std::endl;
+
+        constexpr char printLine[] = "------------------------------------------------------------------------";
+        while(true)
+        {
+            std::cout << printLine << std::endl;
+            if(yesNoPrompt("Capture (y) or finish (n)?"))
+            {
+                std::cout << "Capturing calibration object" << std::endl;
+                const auto detectionResult = Zivid::Experimental::Calibration::detectFeaturePoints(camera);
+                const auto input = Zivid::Experimental::Calibration::InfieldCorrectionInput{ detectionResult };
+
+                if(input.valid())
+                {
+                    dataset.emplace_back(input);
+                    std::cout << "Valid measurement at: " << input.detectionResult().centroid() << std::endl;
+                }
+                else
+                {
+                    std::cout << "****INVALID****" << std::endl;
+                    std::cout << "Feedback: " << input.statusDescription() << std::endl;
+                }
+                std::cout << printLine << std::endl;
+            }
+            else
+            {
+                std::cout << "End of capturing stage." << std::endl;
+                std::cout << printLine << std::endl;
+                break;
+            }
+            std::cout << "You have collected " << dataset.size() << " valid measurements so far." << std::endl;
+        }
+        return dataset;
+    }
+
+} // namespace
+
+int main()
+{
+    try
+    {
+        Zivid::Application zivid;
+        std::cout << "Connecting to camera" << std::endl;
+        auto camera = zivid.connectCamera();
+
+        // Gather data
```

Comment:
We use present continuous tense (ing) for comments and print outs as a standard. We tend to print rather than comment, only for details we comment rather than print.

Reply:
Changing to an appropriate print.